src/msg/async/ProtocolV1.cc

```diff
@@ -1320,22 +1324,22 @@ void ProtocolV1::reset_recv_state()
   }
 }
 
-Message *ProtocolV1::_get_next_outgoing(ceph::buffer::list *bl) {
-  Message *m = 0;
+ProtocolV1::out_q_entry_t ProtocolV1::_get_next_outgoing(ceph::buffer::list *bl) {
```

Comment:
The interface of this method became weird – the bufferlist is returned twice: 1) as a part of `out_q_entry_t` and 2) as the `bl` output parameter.

Reply:
@rzarzynski would you prefer if I return a pair of the message and is_prepared?

Reply:
I think returning the struct is fine; I would just try kill the `bl` output parameter.

Reply:
Got it, thanks!